### PR TITLE
#9392 Added possibility to print qr code

### PR DESCRIPTION
**Pull request by MrTschi**

### CODE REVIEW

**Comment by matiu** [comment]
thanks a lot for the contribution! we will review it soon!

File: src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.html [hunk 1]
```diff
@@ -84,10 +84,20 @@
...
     <ion-list *ngSwitchCase="'qr-code'">
       <div class="qrCode" *ngIf="supported && showQrCode">
+        <div class="on-print-show">
+          <p translate>QR code to recover wallet {{walletName}}</p>
+        </div>
         <ngx-qrcode qrc-value="{{exportWalletInfo}}" qrc-class="aclass" qrc-errorCorrectionLevel="M"></ngx-qrcode>
         <ion-item-divider>
           <span translate>From the destination device, go to Add wallet &gt; Import wallet and scan this QR code</span>
         </ion-item-divider>
+        <ion-item-divider class="on-print-hide">
```
Comment (Gamboster) [inline src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.html]: You could add *ngIf="!isCordova" here.

**Comment by MrTschi** [comment]
Thank you for the feedback guys. I‘ll have a look into within the next days if I‘ll find a solution for android. 
If not, i‘ll add the limitation suggested by @Gamboster

**Comment by MrTschi** [comment]
I committed a preliminary state of my work, will finalize it on weekend. Could you please give me feedback to the pullrequest?
I‘m not sure regarding the document.getElementById implementation, is there a way to do it better?

Tested it in browser and on iOS, both are working. 

af4aa6ed-1df0-4b4a-9f7a-b10389956334

File: src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.ts [hunk 2]
```diff
@@ -363,6 +372,16 @@ export class WalletExportPage extends WalletTabsChild {
     });
   }
 
+  public printQr(): boolean {
+    if (this.isCordova) {
+      let printContent = document.getElementById('cordovaPrintWrapper');
```
Comment (Gamboster) [inline src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.ts]: It should be `const` instead `let`

File: src/providers/index.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /* Native modules */
 export { AndroidFingerprintAuth } from '@ionic-native/android-fingerprint-auth';
 export { Clipboard } from '@ionic-native/clipboard';
+export { Printer } from '@ionic-native/printer';
```
Comment (Gamboster) [inline src/providers/index.ts]: Please, in alphabetical order

File: package-lock.json [hunk 4]
```diff
@@ -557,6 +557,11 @@
       "resolved": "https://registry.npmjs.org/@ionic-native/launch-review/-/launch-review-4.14.0.tgz",
       "integrity": "sha512-8Yl0r+RmEoaCulccUvLjWDsTw2MV2dKMLccvcOfzOQv2iJnnnBMQ43JsFOxX02ufRc4V5OafPqYGNAN+Krxisw=="
     },
+    "@ionic-native/printer": {
```
Comment (Gamboster) [inline package-lock.json]: Do not commit the package-lock file

File: package.json [hunk 5]
```diff
@@ -167,6 +167,7 @@
     "@ionic-native/fcm": "4.14.0",
     "@ionic-native/file": "4.14.0",
     "@ionic-native/launch-review": "4.14.0",
+    "@ionic-native/printer": "4.18.0",
```
Comment (Gamboster) [inline package.json]: Use the same version that we are using for other plugins, please (4.14.0)

File: src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.scss [hunk 6]
```diff
@@ -7,9 +7,23 @@ page-wallet-export {
...
     }
+    .on-print-show {
+      display: none;
+    }
   }
   .not-supported-message {
     margin-top: 5rem;
     text-align: center;
   }
+  @media print {
```
Comment (Gamboster) [inline src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.scss]: Try to center the image and text when print in mobile

**Comment by Gamboster** [comment]
I addition, you could add an option to `printer.print(content, options)` function, and set a name for the file generated. Maybe something like this:
```let options: PrintOptions = {
     name: 'Exported wallet from ' + appName
   };```
To prevent the "unknown" name.

Also, you can add some logger, in `isAvailable().then()`

File: src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.scss (hunk 6 above)
Comment (MrTschi) [reply src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.scss]: The CSS is only available when printing on desktop. Within cordova, only the div element <div id="cordovaPrintWrapper"> is passed to the printer. If I'm going to pass the whole <html> or <body>, the printing preview on the device is showing an empty page.

**Comment by Gamboster** [comment]
@MrTschi 
As a security policy, we can not merge commits that have not been signed.
Could you do that last thing? It even seems that you have sent commits from different users
_9392_added_possibility_to_print_qr_code_by_mrtschi_ _pull_request__9393_ _bitpay_copay

**Comment by MrTschi** [comment]
Hi guys, thanks for the feedback!
I rebased my commits to sign them and to updated commit author to the correct one:
```
git commit -S --amend --author "MrTschi <______@icloud.com>" --no-edit && git rebase --continue
```
Finally I force-pushed it.
I also updated the CONTRIBUTION.md and added the fact that a commit must be signed.

**Pull request merged**